pennylane/operation.py

```diff
@@ -694,19 +694,38 @@ def _primitive_bind_call(cls, *args, **kwargs):
         if cls._primitive is None:
             # guard against this being called when primitive is not defined.
             return type.__call__(cls, *args, **kwargs)
-        iterable_wires_types = (list, tuple, qml.wires.Wires, range, set)
+
+        import jax  # pylint: disable=import-outside-toplevel
+
+        array_types = (jax.numpy.ndarray, np.ndarray)
+        iterable_wires_types = (
+            list,
+            tuple,
+            qml.wires.Wires,
+            range,
+            set,
+            *array_types,
+        )
 
         # process wires so that we can handle them either as a final argument or as a keyword argument.
         # Stick `n_wires` as a keyword argument so we have enough information to repack them during
         # the implementation call defined by `primitive.def_impl`.
         if "wires" in kwargs:
             wires = kwargs.pop("wires")
-            wires = tuple(wires) if isinstance(wires, iterable_wires_types) else (wires,)
+            if isinstance(wires, array_types) and wires.shape == ():
+                wires = (wires,)
+            elif isinstance(wires, iterable_wires_types):
+                wires = tuple(wires)
+            else:
+                wires = (wires,)
             kwargs["n_wires"] = len(wires)
             args += wires
+        elif args and isinstance(args[-1], array_types) and args[-1].shape == ():
```

Comment:
```suggestion
        # If not in kwargs, check if the last positional argument represents wire(s).
        elif args and isinstance(args[-1], array_types) and args[-1].shape == ():
```

might be helpful to understand the logic